### PR TITLE
blog: add release post for 8.0.0

### PR DESCRIPTION
**Pull request by jasnell**
@cjihrig @mcollina @nodejs/ctc @ZibbyKeaton ... here's the draft. PTAL

### CODE REVIEW

File: locale/en/blog/release/v8.0.0.md [hunk 1]
```diff
@@ -0,0 +1,1216 @@
...
+category: release
+title: Node v8.0.0 (Current)
+slug: node-v8-0-0
+layout: blog-post.hbs
+author: James M Snell, Colin Ihrig
+---
+
+The next major release of Node.js brings a range of significant changes and
+additions, far too many for us to cover adequately in a blog post such as this.
+This article contains a summary of the changes and details we feel are the most
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: Simplify:

> This article contains a summary of the most significant changes and features.

File: locale/en/blog/release/v8.0.0.md [hunk 2]
```diff
@@ -0,0 +1,1216 @@
...
+In Node.js 6.0.0, a new set of `Buffer` construction methods
+[were introduced](https://medium.com/@jasnell/node-js-buffer-api-changes-3c21f1048f97)
+as an alternative to calling the `Buffer(num)` constructor in order to address a
+number of [security](https://nodesecurity.io/advisories/67) and usability
+concerns. The existing constructor, however, is used extensively throughout the
+Node.js ecosystem, making it infeasible for us to fully deprecate or disable it
+without causing significant breakage.
+
+Zero-filling new instances of `Buffer(num)` by default will have a noticeable
+and significant impact on application performance. To mitigate the loss in
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: `noticeable and significant` -> `significant`; `application performance` -> `performance`

File: locale/en/blog/release/v8.0.0.md [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+performance, developers should move to the new `Buffer.allocUnsafe(num)` API if
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: Remove `To mitigate the loss in performance`

File: locale/en/blog/release/v8.0.0.md [hunk 4]
```diff
@@ -0,0 +1,1216 @@
...
+// Zero-filled Buffers
+const safeBuffer1 = Buffer.alloc(10);
+const safeBuffer2 = new Buffer(10);
+
+// Uninitialized Buffer
+const unsafeBuffer = Buffer.allocUnsafe(10);
+```
+
+Note that while the `Buffer(num)` constructor will not be removed from Node.js,
+its continued use is considered to be unsupported.
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: At a minimum: `considered to be unsupported` -> `deprecated`

But really, I'd say remove this sentence altogether. We can't truly guarantee that it will not be removed from Node.js ever (although I agree it's unlikely).

File: locale/en/blog/release/v8.0.0.md [hunk 5]
```diff
@@ -0,0 +1,1216 @@
...
+
+### Pending Deprecations
+
+To make it easier to catch uses of `Buffer(num)` within an application at
+development time, or within CI testing environments, a new
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: remove comma after `time` (but leave the one after `environments`)

File: locale/en/blog/release/v8.0.0.md [hunk 6]
```diff
@@ -0,0 +1,1216 @@
...
+
+### Debugger changes
+
+The legacy command line debugger is being removed in Node.js 8. The underlying
+debugging protocol has been deprecated in V8 for several years now, and will
+finally be removed. Therefore, Node.js is removing it completely. As a command
+line replacement, `node-inspect` has been integrated directly into the Node.js
+runtime. Additionally, the V8 Inspector debugger, which arrived previously as
+an experimental feature in Node.js 6, is being upgraded to an fully supported
+feature.
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: `an fully supported` -> `a fully supported`

I think this paragraph is a fair bit too wordy. Something succinct might be better. Maybe drop a couple sentences so it looks like this?:

> The legacy command line debugger is being removed in Node.js 8. As a command
> line replacement, `node-inspect` has been integrated directly into the Node.js
> runtime. Additionally, the V8 Inspector debugger, which arrived previously as
> an experimental feature in Node.js 6, is being upgraded to a fully supported
> feature.

File: locale/en/blog/release/v8.0.0.md [hunk 7]
```diff
@@ -0,0 +1,1216 @@
...
+[Native Abstractions for Node.js (`nan`)](https://www.npmjs.com/package/nan)
+that will allow native addons to be compiled once on a system and used across
+multiple versions of Node.js.
+
+By providing a new virtual machine agnostic Application Binary Interface (ABI),
+it becomes possible for native addons to work not only against multiple
+versions of the V8 JavaScript runtime, but Microsoft's Chakra-Core runtime
+as well.
+
+The [N-API](https://nodejs.org/api/n-api.html) is still considered experimental
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: Remove `considered`. (It's either experimental or it's not. Rather than provide clarity, 'considered' just muddies things.)

File: locale/en/blog/release/v8.0.0.md [hunk 8]
```diff
@@ -0,0 +1,1216 @@
...
+// Listen for inspector events
+session.on(‘inspectorNotification’, (message) => /** ... **/);
+
+// Send messages to the inspector
+session.post(message);
+
+session.disconnect();
+```
+
+Note that the inspector API is still considered experimental and may change
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: remove `still considered`

File: locale/en/blog/release/v8.0.0.md [hunk 9]
```diff
@@ -0,0 +1,1216 @@
...
+significantly.
+
+### Long Term Support
+
+As an even numbered release, Node.js 8.0.0 represents the next release line to
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: Remove `As an even numbered release,`

Nit: `represents` -> `is`

File: locale/en/blog/release/v8.0.0.md [hunk 10]
```diff
@@ -0,0 +1,1216 @@
...
+enter Long Term Support (LTS). This is scheduled to happen in October, once the
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: `October` -> `October 2017`

File: locale/en/blog/release/v8.0.0.md [hunk 11]
```diff
@@ -0,0 +1,1216 @@
...
+Node.js 8 release line has had several months to completely stabilize. Once
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: Remove `once the Node.js 8 release line has had several months to completely stabilize`

File: locale/en/blog/release/v8.0.0.md [hunk 12]
```diff
@@ -0,0 +1,1216 @@
...
+Node.js 8 transitions to LTS, it will receive the code name Carbon.
+
+![Node.js Long Term Support Schedule](https://talks.continuation.io/nodeweek-4-17/images/lts.png)
+
+If you’ve been using Node.js for any significant amount of time, it is worth
+noting that we’ve also dropped the “v” in Node.js 8. Previous versions  were
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: Nit: Change first sentence in this paragraph to `Note that we have dropped the "v" in Node.js 8.`

File: locale/en/blog/release/v8.0.0.md [hunk 13]
```diff
@@ -0,0 +1,1216 @@
...
+commonly referred to as v0.10, v0.12, v4, v6, etc. In order to avoid confusion
+with V8, the underlying JavaScript engine, we’ve decided to simply drop the “v”
```
Comment (Trott) [inline locale/en/blog/release/v8.0.0.md]: `we've decided to simply drop` -> `we've dropped`

**Review by Trott — COMMENTED** [review]
Left a bunch of nits, but overall: Wow, this is a lot to summarize and you've organized it well. :+1:

**Review by mcollina — APPROVED** [review]

**Review by cjihrig — APPROVED** [review]

**Review by amandeepmittal — APPROVED** [review]

**Comment by jasnell** [comment]
Updated to address the comments. Will update again once the release is out.

**Review by mhdawson — APPROVED** [review]
LGTM

File: locale/en/blog/release/v8.0.0.md [hunk 14]
```diff
@@ -0,0 +1,1210 @@
...
+
+```js
+const fs = require('fs');
+const util = require('util');
+
+const readfile = util.promisify(fs.readFile);
+
+readfile('/some/file')
+  .then((data) => { /** ... **/ })
+  .catch((err) => { /** ... **/ }) ;
```
Comment (vsemozhetbyt) [inline locale/en/blog/release/v8.0.0.md]: A nit: unnecessary space before semicolon.

File: locale/en/blog/release/v8.0.0.md [hunk 15]
```diff
@@ -0,0 +1,1210 @@
...
+feature.
+
+### Experimental inspector JavaScript API
+
+A new experimental JavaScript API for the Inspector protocol has been introduced
+enabling developers new ways of leveraging the debug protocol to inspect running
+Node.js processes.
+
+```js
+const inspector = require(‘inspector’);
```
Comment (vsemozhetbyt) [inline locale/en/blog/release/v8.0.0.md]: Wrong quotes, parsing error.

File: locale/en/blog/release/v8.0.0.md [hunk 16]
```diff
@@ -0,0 +1,1210 @@
...
+
+const session = new inspector.Session();
+session.connect();
+
+// Listen for inspector events
+session.on(‘inspectorNotification’, (message) => /** ... **/);
```
Comment (vsemozhetbyt) [inline locale/en/blog/release/v8.0.0.md]: 1. Wrong quotes, parsing error.
2. Maybe `(message) => { /** ... **/ }` ?

Comment (drewfish) [reply locale/en/blog/release/v8.0.0.md]: also funny quotes on `require(‘inspector’)`

**Comment by jasnell** [comment]
Landed with fixes made. Thank you all

**Comment by lpinca** [comment]
@jasnell I think you only pushed on your fork.

**Comment by jasnell** [comment]
I need a beer. fixing

**Comment by vsemozhetbyt** [comment]
Sorry, is this OK:

> Information for static error codes can be queried quickly by referencing the Node.js documentation. For instance, the URL to lookup information about the ERR_ASSERTION error code is https://nodejs.org/dist/latest-v7.x/docs/api/errors.html#ERR_ASSERTION.

Should this be `...latest-v7.x...`? And there is no ERR_ASSERTION still in the https://github.com/nodejs/node/blob/master/doc/api/errors.md, so the link is wrong both for v7 and v8.

**Comment by vsemozhetbyt** [comment]
> Processing warnings such as deprecations may now to redirected to a file

to redirected -> be redirected ?

**Comment by vsemozhetbyt** [comment]
> The npm client has been updated to 5.0.0 [3c3b36af0f] #12936.

These are links for the beta, maybe they should be the https://github.com/nodejs/node/commit/c58cea5a163cd5d7133e00fdf257325ce3807c09 and https://github.com/nodejs/node/pull/13276 instead.

**Comment by vsemozhetbyt** [comment]
> NODE_MODULE_VERSION has been updated to 57 (https://github.com/nodejs/node/commit/ec7cbaf266)] #12995.

Wrong link formatting, wrong (not shortened) rendering in the https://nodejs.org/en/blog/release/v8.0.0/

See also https://github.com/nodejs/node/pull/13313

**Comment by vsemozhetbyt** [comment]
Maybe it is too late for this, but should not we mention in notable changes some `cli` changes versus v6, like:

https://github.com/nodejs/node/pull/12028
https://github.com/nodejs/node/pull/13012
https://github.com/nodejs/node/pull/10651

File: locale/en/blog/release/v8.0.0.md (hunk 16 above)
Comment (vsemozhetbyt) [reply locale/en/blog/release/v8.0.0.md]: It seems they are also fixed)

**Comment by vsemozhetbyt** [comment]
@jasnell , @Trott , @nodejs/documentation
Sorry for being so late. Can we still fix the last nits? I am not sure about some of them and about the procedure now.

**Comment by Trott** [comment]
> I am not sure about some of them and about the procedure now.

Procedure might be to open a pull request to make the changes. /cc @nodejs/website in case I'm wrong about that...

**Comment by vsemozhetbyt** [comment]
@Trott PTAL: https://github.com/nodejs/nodejs.org/pull/1244

**Comment by jasnell** [comment]
@vsemozhetbyt ... no problems :-) we can make edits to the blog post. Just add commits to the PR I see you've already opened :-)

File: locale/en/blog/release/v8.0.0.md [hunk 17]
```diff
@@ -0,0 +1,1292 @@
...
+[were introduced](https://medium.com/@jasnell/node-js-buffer-api-changes-3c21f1048f97)
+as an alternative to calling the `Buffer(num)` constructor in order to address a
+number of [security](https://nodesecurity.io/advisories/67) and usability
+concerns. The existing constructor, however, is used extensively throughout the
+Node.js ecosystem, making it infeasible for us to fully deprecate or disable it
+without causing significant breakage.
+
+Zero-filling new instances of `Buffer(num)` by default will have a significant
+impact on performance. Developers should move to the new
+`Buffer.allocUnsafe(num)` API if they wish to allocate `Buffer` instances
```
Comment (ChALkeR) [inline locale/en/blog/release/v8.0.0.md:109]: This probably should have also mentioned `Buffer.alloc` (for consistent behaviour between versions), and an additional warning against using uninitialized memory allocations.

@jasnell, would it be acceptable to update the release post now? If yes, I will be willing to propose a small patch.
